providers/amazon/src/airflow/providers/amazon/aws/operators/ecs.py

```diff
@@ -517,7 +517,7 @@ def execute(self, context):
         if self.reattach:
             # Generate deterministic UUID which refers to unique TaskInstanceKey
             ti: TaskInstance = context["ti"]
-            self._started_by = generate_uuid(*map(str, ti.key.primary))
+            self._started_by = generate_uuid(*map(str, [ti.dag_id, ti.map_index, ti.run_id, ti.task_id]))
```

Comment:
Does this work for both Airflow 2 and 3?
Asking as the issue reported problems only with Airflow 3 so I wonder if the fix is for both or needs to be under version flag?

Reply:
It should be backwards compatible, since a true TaskInstance model object also has those fields (which were previously being fetch from the primary key). But I can do some manual UAT to double check.

Reply:
Tested to ensure compatibility. See the updated details section.